statistics/semantic/src/main/java/com/spotify/heroic/statistics/semantic/SemanticApiReporter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2017 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.statistics.semantic;
+
+import com.codahale.metrics.Histogram;
+import com.spotify.heroic.statistics.ApiReporter;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricRegistry;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@ToString(of = {"base"})
+public class SemanticApiReporter implements ApiReporter {
+    private static final String COMPONENT = "api";
+    private static final long SLI_SMALL_QUERY_THRESHOLD = 200000;
+
+    private final Histogram sliQueryLatencyForSmallQueries;
+
+    public SemanticApiReporter(SemanticMetricRegistry registry) {
+        final MetricId base = MetricId.build().tagged("component", COMPONENT);
+
+        sliQueryLatencyForSmallQueries = registry.histogram(
+            base.tagged("what", "sli-query-latency-for-small-queries", "unit", Units.MILLISECOND));
+    }
+
+    @Override
+    public void reportCompletedQuery(long sampleSize, long duration) {
+        if (sampleSize < SLI_SMALL_QUERY_THRESHOLD) {
```

Comment:
This doesn't seem like the right location to decide this. The call site seems more natural.
Also, `SLI_SMALL_QUERY_THRESHOLD` should probably be configurable, since its bound to be tweaked.